applications/serial_lte_modem/src/slm_at_commands.c

```diff
@@ -216,31 +216,25 @@ static int handle_at_modemreset(enum at_cmd_type type)
 	unsigned int step = 1;
 	int ret;
 
-	do {
-		ret = nrf_modem_lib_shutdown();
-		if (ret != 0) {
-			break;
-		}
-		++step;
 
-		ret = nrf_modem_lib_init();
-		if (ret < 0) {
-			break;
-		}
-		++step;
+	ret = nrf_modem_lib_shutdown();
+	if (ret != 0) {
+		goto out;
+	}
+	++step;
 
-		if (ret > 0 || (slm_fota_stage == FOTA_STAGE_ACTIVATE
-					&& (slm_fota_type == DFU_TARGET_IMAGE_TYPE_MODEM_DELTA ||
-					    slm_fota_type == DFU_TARGET_IMAGE_TYPE_FULL_MODEM))) {
-			slm_finish_modem_fota(ret);
-			slm_fota_post_process();
-		}
+	ret = nrf_modem_lib_init();
 
-		/* Success. */
-		rsp_send("\r\n#XMODEMRESET: 0\r\n");
-		return 0;
-	} while (0);
+	if ((slm_fota_stage == FOTA_STAGE_ACTIVATE
+				&& (slm_fota_type == DFU_TARGET_IMAGE_TYPE_MODEM_DELTA ||
+					slm_fota_type == DFU_TARGET_IMAGE_TYPE_FULL_MODEM))) {
+#if defined(CONFIG_SLM_FULL_FOTA)
+		slm_finish_modem_full_dfu(ret);
+#endif
+		slm_fota_post_process();
+	}
```

Comment:
There seems to be some issue somewhere here. In SLM delta modem fota tests, "#XFOTA: 5,0" is missing after AT#XMODEMRESET.

Reply:
Thank you, yes, as the dfu_handler is called before nrf_modem_lib_init returns, the slm_fota_stage is changed. I've added a fix to get the fota stage before performing the modem reinitialization.

Reply:
It seems to work now with XMODEMRESET

Reply:
@tomi-font I didn't want to use `FOTA_STAGE_COMPLETE` as I don't know when it is cleared. So if the application were to call XMODEMRESET multiple times, it would send the XFOTA event multiple times, although there were only one DFU.

Reply:
@VTPeltoketo Thank you for checking!